crawl-ref/source/stairs.cc

```diff
@@ -529,6 +552,26 @@ static level_id _travel_destination(const dungeon_feature_type how,
 }
 
 /**
+ * Check to see if the level transition should be stopped
+ *
+ * @param dest      The destination level (branch and depth).
+ * @param feat      The dungeon feature the player is standing on.
+ * @param going_up  True if the player is trying to go up stairs.
+ * @return          True if the level transition should be stopped.
+ */
+static bool _stop_transition(level_id dest, dungeon_feature_type feat,
+                             bool going_up)
+{
```

Comment:
I made this a new function to try to clarify this logic since it seems a little clunky.  Is it better to leave in take_stairs()? I like it separate since it's easier to understand the overall flow in take_stairs()

Reply:
Seems fine to me!